### ISSUE TITLE
iOS app/mobile Safari displays "invalid timestamp"

### ISSUE THREAD

**Issue by gvssr** [pippyn/Home-Assistant-Sensor-Afvalbeheer#293]
I have 3 sensors setup with this integration. However, on the iOS apps and mobile safari they display as “invalid timestamp”, while they display correctly on desktop browsers.

Proper display on desktop:
![image](https://user-images.githubusercontent.com/61377476/149922780-edc96c95-316a-406f-b89e-10f9e0a15eac.png)

Incorrect display on mobile:
![image](https://user-images.githubusercontent.com/61377476/149923097-de0cb129-c1d4-4b8b-8a64-51fbad0148d4.jpeg)

I first reported this as a Home Assistant frontend issue here: https://github.com/home-assistant/frontend/issues/11351 but apparently it’s an ISO 8601 timestamp issue and up to the integration author to fix, according to this comment:

> Apple strongly enforces ISO 8601 compliant timestamps, meaning the "T" between the data and time need to be there.

> See the warning here: https://www.home-assistant.io/docs/configuration/templating/

> Where are those entities from? Are they template entities that you created? If yes, you can adjust the timestamp yourself (see link above), otherwise the integration author should take care of that.

**Comment by gvssr** [comment]
Additional info, straight from Home Assistant docs: https://www.home-assistant.io/docs/configuration/templating/
![19ED9F7B-E91F-4B07-9E36-C7B7A9DF2455](https://user-images.githubusercontent.com/61377476/150015136-af9ae388-fa9d-4fa9-b2cc-cbbaa9485884.jpeg)

This integration currently seems to omit the “T”. Can that be added, to be compliant with HA standards?
![01BA33D9-B06B-4F27-A3F8-9DCEDB4A2194](https://user-images.githubusercontent.com/61377476/150015427-2abf14bc-4ce1-4e3f-ab12-d482e5b6cc1d.jpeg)

**Comment by pippyn** [comment]
What is your sensor config? Do you use the option to make the sensor state a date object?

**Comment by gvssr** [comment]
@pippyn I believe I have it setup as a date object. Should it be something else? Below is the config:

```
  - platform: afvalbeheer
    upcomingsensor: 1
    wastecollector: HVC
    resources:
      - gft
      - papier
      - pmd
    postcode: XXXXAA
    streetnumber: XX
    dateobject: 1
```

**Comment by pippyn** [comment]
Do you need the date object? If not remove that option and all should be fine

**Comment by gvssr** [comment]
I'd rather it be a dateobject as that way I can show relative time like in the screenshots in my first post ("next week, next month"). Otherwise it's just a string of numbers.

**Comment by gvssr** [comment]
I'd be fine to use a custom dateformat. I've tried '%F' and  '%Y-%m-%dT%H:%M:%S%z' but both seem to still omit the "T"

**Comment by pippyn** [comment]
I am working on make the timestamps ISO compliant, but it is not ready yet.

The problem is not all API's report the timezone, so I'll have to convert them and do a lot of tests.

For now you'll have to sit tight i'm afraid.

**Comment by gvssr** [comment]
No problem! I’m glad this integration exists at all, happy to wait. I’ll see if I can figure out making a template sensor for each that’s ISO compliant.

**Comment by pippyn** [comment]
Yes this problem only exists with the dateobject option. Most people won't need that option though.
And there is no GUI yet, I'll start building that soon.
